src/storage/backend/backend_impl.rs

```diff
@@ -98,21 +98,29 @@ impl Storage for Backend {
     async fn load_from_self(&self, ino: INum, block_id: usize) -> StorageResult<Option<Block>> {
         let mut block = Block::new_zeroed(self.block_size);
 
-        if let Err(e) = self
-            .operator
-            .reader(&get_block_path(ino, block_id))
-            .await?
-            .read(block.make_mut_slice())
-            .await
-        {
-            if e.kind() == std::io::ErrorKind::NotFound {
-                Ok(None)
-            } else {
-                Err(e.into())
+        let mut reader = self.operator.reader(&get_block_path(ino, block_id)).await?;
+        // Check if the reader point is at the end of the file.
+        loop {
+            match reader.read(block.make_mut_slice()).await {
```

Comment:
call `make_mut_slice()`  outside the loop ,it is not zero cost

Reply:
The block is just created in this method, and the reference count of it should always be 1, therefore `make_mut_slice` should not perform any copy, and use it in the loop is OK.

However, the start point of the block slice for `reader.read` to write is always 0 in this case, and when the loop continues, it will overwrite the contents which are readed in the last iteration. You should set a varible to record the start point in each iteration.